docs/content/basic/authenticating-oauth.md

```diff
@@ -4,256 +4,703 @@ lang: en
+up OAuth routes, state verification, and passing your app an installation object
+which you must store.
+
+### App options
+
+The following `App` options are required for OAuth installations:
 
-## Development and Testing
+- `clientId`: unique identifier of the application client.
+- `clientSecret`: secret value to confirm the client ID.
+- `stateSecret`: secret value used for [state verificiation][verification] of
+  authentication requests.
+- `scopes`: permissions requested for the `bot` user during installation.
+  [Explore scopes][scopes].
+- `installationStore`: handlers that store, fetch, and delete installation
+  information.
 
-We've provided a default implementation of the `installationStore`  `FileInstallationStore` which you can use during app development and testing.
+#### Development and testing
+
+Here we've provided a default implementation of the `installationStore` with
+[`FileInstallationStore`][store-file-installation] which can be useful when
+developing and testing your app:
 
 ```javascript
-const { App } = require('@slack/bolt');
-const { FileInstallationStore } = require('@slack/oauth');
+const { App } = require("@slack/bolt");
+const { FileInstallationStore } = require("@slack/oauth");
 const app = new App({
   signingSecret: process.env.SLACK_SIGNING_SECRET,
   clientId: process.env.SLACK_CLIENT_ID,
   clientSecret: process.env.SLACK_CLIENT_SECRET,
-  stateSecret: 'my-state-secret',
-  scopes: ['channels:history', 'chat:write', 'commands'],
+  stateSecret: "my-state-secret",
+  scopes: ["channels:history", "chat:write", "commands"],
   installationStore: new FileInstallationStore(),
 });
 ```
-:::warning 
 
-This is **_not_** recommended for use in production - you should implement your own production store. Please see the example code to the right and [our other examples](https://github.com/slackapi/bolt-js/tree/main/examples/oauth).
+:::warning
+
+This is **not** recommended for use in production - you should
+[implement your own installation store](#installation-store). Please continue
+reading or inspect [our example apps][examples].
 
 :::
 
+### Installer options
+
+We provide several options for customizing default OAuth using the
+`installerOptions` object, which can be passed in during the initialization of
+`App`. You can override these common options and
+[find others here][install-provider-options]:
+
+- `authVersion`: Settings for either new Slack Apps (`v2`) or Classic Slack Apps
+  (`v1`). Default: `v2`.
+- `directInstall`: Skip the [default installation page][installation] at
+  `installPath`. Default: `false`.
+- `installPath`: Path of the URL for starting an installation. Default:
+  `/slack/install`.
+- `metadata`: Relevant session information passed between requests. Optional.
+- `redirectUriPath`: Path of the installation callback URL. Default:
+  `/slack/oauth_redirect`.
+- `stateVerification`: Option to skip state verification for requests. Default:
```

Comment:
how about having type as well? it would make this part even easier to understand
```suggestion
- `stateVerification: boolean`: Option to skip state verification for requests. Default:
```

Reply:
Awesome callout! I made a small update to formatting, but this makes this reference so much better 🙏